cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/GoalOptimizer.java

```diff
@@ -494,10 +495,7 @@ public OptimizerResult optimizations(ClusterModel clusterModel,
    */
   public Set<String> excludedTopics(ClusterModel clusterModel, Pattern requestedExcludedTopics) {
     Pattern topicsToExclude = requestedExcludedTopics != null ? requestedExcludedTopics : _defaultExcludedTopics;
-    return clusterModel.topics()
-                       .stream()
-                       .filter(topic -> topicsToExclude.matcher(topic).matches())
-                       .collect(Collectors.toSet());
+    return Utils.getTopicNamesMatchedWithPattern(topicsToExclude, clusterModel.topics());
```

Comment:
(Applies to other similar uses of `getTopicNamesMatchedWithPattern`) If `topicsToExclude.pattern()` is empty, then this function should ideally terminate early w/o performing redundant operations.

-- Currently regardless of whether the pattern is empty or not, `clusterModel.topics()` is called, which is an expensive call that generates objects for no reason. Similarly, call to `getTopicNamesMatchedWithPattern` is expensive with redundant `Matcher` generation. All of these can be eliminated.

(Can we also add a unit test to verify that behavior?)